### PR TITLE
fix(PE-7080): add apis for fetching paginated delegates

### PR DESCRIPTION
**Pull request by dtfiedler**
TODO: 
- apis for updating delegates (ref: https://github.com/ar-io/ar-io-network-process/blob/develop/src/main.lua#L2913-L3009)

### CODE REVIEW

File: src/common/io.ts [hunk 1]
```diff
@@ -313,6 +287,40 @@ export class IOReadable implements AoIORead {
     });
   }
 
+  async getGatewayDelegates(
+    params?: { address: WalletAddress } & PaginationParams,
```
Comment (arielmelendez) [inline src/common/io.ts:291]: Would be interesting if the paginationparams type were genericized to allow for specifying the allowable sortBy values for each use case.

**Review by fedellen — DISMISSED** [review]

File: src/types/io.ts [hunk 2]
```diff
@@ -312,6 +315,14 @@ export interface AoIORead {
   }: {
     address: WalletAddress;
   }): Promise<AoGateway | undefined>;
+  // TODO: these could be moved to a separate Gateways class that implements gateway specific interactions
+  getGatewayDelegates(
+    params?: PaginationParams,
```
Comment (arielmelendez) [inline src/types/io.ts:320]: The input type here is different the actual implementation (where `address` is unioned into the type). Same below.

File: tests/unit/utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      { name: 'Tag1', value: undefined },
+      { name: 'Tag2', value: undefined },
+    ];
+
+    const prunedTags = pruneTags(tags);
+
+    assert.deepEqual(prunedTags, []);
+  });
+
+  it('should return same array when no tags have undefined values', () => {
```
Comment (arielmelendez) [inline tests/unit/utils.test.ts:34]: Could add an empty tags test as well.

Comment (dtfiedler) [reply tests/unit/utils.test.ts:34]: [7a1abc4](https://github.com/ar-io/ar-io-sdk/pull/248/commits/7a1abc4869aed61a39cd99a313c64aff7c5e10de)

**Review by fedellen — APPROVED** [review]

**Review by arielmelendez — APPROVED** [review]

**Pull request merged**